dashboard/app/models/game.rb

```diff
@@ -216,78 +216,89 @@ def channel_backed?
     [APPLAB, GAMELAB, WEBLAB, SCRATCH, PIXELATION].include? app
   end
 
+  # Format: name:app:intro_video
+  # Don't change the order of existing entries! Always append to the end of the list.
+  GAMES_BY_INDEX = %w(
+    Maze:maze:maze_intro
+    Artist:turtle:artist_intro
+    Artist2:turtle
+    Farmer:maze:farmer_intro
+    Artist3:turtle
+    Farmer2:maze
+    Artist4:turtle
+    Farmer3:maze
+    Artist5:turtle
+    MazeEC:maze:maze_intro
+    Unplug1:unplug
+    Unplug2:unplug
+    Unplug3:unplug
+    Unplug4:unplug
+    Unplug5:unplug
+    Unplug6:unplug
+    Unplug7:unplug
+    Unplug8:unplug
+    Unplug9:unplug
+    Unplug10:unplug
+    Unplug11:unplug
+    Bounce:bounce
+    Custom:turtle
+    Flappy:flappy:flappy_intro
+    CustomMaze:maze
+    Studio:studio
+    Jigsaw:jigsaw
+    MazeStep:maze
+    Multi:multi
+    Match:match
+    Unplugged:unplug
+    Wordsearch:wordsearch
+    CustomStudio:studio
+    Calc:calc
+    Webapp:webapp
+    Eval:eval
+    ArtistEC:turtle:artist_intro
+    TextMatch
+    StudioEC:studio
+    ContractMatch
+    Applab:applab
+    NetSim:netsim
+    External:external
+    Pixelation:pixelation
+    TextCompression:text_compression
+    Odometer:odometer
+    FrequencyAnalysis:frequency_analysis
+    Vigenere:vigenere
+    Craft:craft
+    Gamelab:gamelab
+    LevelGroup:level_group
+    FreeResponse:free_response
+    NotUsed:not_used
+    StandaloneVideo:standalone_video
+    ExternalLink:external_link
+    EvaluationMulti:evaluation_multi
+    PublicKeyCryptography:public_key_cryptography
+    Weblab:weblab
+    CurriculumReference:curriculum_reference
+    Map:map
+    CustomFlappy:flappy
+    Scratch:scratch
+  )
+
   def self.setup
+    games_by_index, intro_video_keys = GAMES_BY_INDEX.inject([[], []]) do |memo, line|
+      games_by_index, intro_video_keys = memo
+      name, app, intro_video_key = line.split ':'
+      games_by_index << [name, app, intro_video_key]
+      intro_video_keys << intro_video_key unless intro_video_key.nil?
+      [games_by_index, intro_video_keys]
+    end
+    videos_by_key = Video.where(key: intro_video_keys).index_by(&:key)
+    games = games_by_index.map.with_index(1) do |game, id|
+      name, app, intro_video_key = game
+      {id: id, name: name, app: app, intro_video_id: videos_by_key[intro_video_key]&.id}
+    end
```

Comment:
This code is quite a bit more complicated than it was previously. Is there any way we can simplify this, or is the extra complexity really necessary to build up the necessary data in memory compared to calling `Video.find_by_key` as before?

Reply:
Yeah, I don't love this.  I was trying to avoid the per-game query _and_ only load the handful of videos we need into the videos_by_key index, but perhaps `Videos.all.index_by(&:key)` is fast enough and would simplify the rest of the logic significantly.  I'll do that.

Reply:
```
[development] dashboard > Benchmark.measure { Video.all.index_by(&:key)}.total
  Video Load (5.9ms)  SELECT `videos`.* FROM `videos` ORDER BY `videos`.`key` ASC
=> 0.013482000000000022
```

:+1: